api/src/main/java/org/openmrs/module/metrics/api/db/hibernate/HibernateMetricsDAO.java

```diff
@@ -54,28 +68,28 @@ public Integer getEncounterObjectsByGivenDateRangeAndType(Date startRange, Date
 	}
 	
 	@Override
-	public Integer getNewPatientsObjectsByGivenDateRange(Date startRange, Date endRange) {
-		// @formatter:off
-		String sql = "select count(*) " + "from metric_event_records e " + "where e.title='Patient' and "
-		        + "(e.date_created between :startRange and :endRange) and " + "e.tags = 'CREATED'";
+	public Integer getNewObjectsByGivenDateRangeAndCategory(Date startRange, Date endRange, String category, String action) {
+		// @formatter:Date
+		String sql = "select count(*) " + "from metric_event_records e " + "where e.category =:category and "
+		        + "(e.date_created between :startRange and :endRange) and " + "e.tags =:action";
```

Comment:
```suggestion
		// @formatter:off
		String sql = "select count(*) " +
					"from metric_event_records e " + 
					"where e.category = :category and " + 
					"(e.date_created between :startRange and :endRange) and " + 
					"e.tags =:action";
		// @formatter:on
```

Reply:
Could we write this in HQL / JPAQL rather than SQL?

```sql
	select count(id)
	from MetricEvent e
	where e.category = :category and
		  e.dateCreated between :startRange and :endRange and
		  e.tags = :action
```

I know this looks basically identical, but the point is to utilise Hibernate where possible rather than raw SQL.